File: examples/demo-go-package/internal/compare/compare.go
package compare

func IsCorrect(response, answer string) bool {
	return response == answer
}

func Feedback(isCorrect bool, correctFeedback, incorrectFeedback string) string {
	if isCorrect {
		if correctFeedback != "" {
			return correctFeedback
		}
		return "Correct"
	}
	if incorrectFeedback != "" {
		return incorrectFeedback
	}
	return "Incorrect"
}

File: examples/demo-rust-package/src/compare.rs
pub fn is_correct(response: Option<&[u8]>, answer: Option<&[u8]>) -> bool {
    response.is_some() && answer.is_some() && response == answer
}

pub fn feedback<'a>(
    is_correct: bool,
    correct_feedback: Option<&'a [u8]>,
    incorrect_feedback: Option<&'a [u8]>,
) -> &'a [u8] {
    if is_correct {
        correct_feedback.unwrap_or(b"Correct!")
    } else {
        incorrect_feedback.unwrap_or(b"Try again.")
    }
}
